core/src/test/java/org/elasticsearch/action/support/replication/ReplicationResponseTests.java

```diff
@@ -117,95 +116,173 @@ public void testFailureEqualsAndHashcode() {
 
-            try (XContentParser parser = xContent.createParser(builder.bytes())) {
-                assertEquals(XContentParser.Token.START_OBJECT, parser.nextToken());
-                assertEquals(XContentParser.Token.FIELD_NAME, parser.nextToken());
-                assertEquals("_shards", parser.currentName());
-                assertEquals(XContentParser.Token.START_OBJECT, parser.nextToken());
+            if (shardInfo.getFailures() != null && shardInfo.getFailures().length > 0) {
                 assertEquals(XContentParser.Token.FIELD_NAME, parser.nextToken());
-                assertEquals("total", parser.currentName());
-                assertEquals(XContentParser.Token.VALUE_NUMBER, parser.nextToken());
-                assertEquals(shardInfo.getTotal(), parser.intValue());
-                assertEquals(XContentParser.Token.FIELD_NAME, parser.nextToken());
-                assertEquals("successful", parser.currentName());
-                assertEquals(XContentParser.Token.VALUE_NUMBER, parser.nextToken());
-                assertEquals(shardInfo.getSuccessful(), parser.intValue());
-                assertEquals(XContentParser.Token.FIELD_NAME, parser.nextToken());
-                assertEquals("failed", parser.currentName());
-                assertEquals(XContentParser.Token.VALUE_NUMBER, parser.nextToken());
-                assertEquals(shardInfo.getFailed(), parser.intValue());
-
-                if (shardInfo.getFailures() != null && shardInfo.getFailures().length > 0) {
-                    assertEquals(XContentParser.Token.FIELD_NAME, parser.nextToken());
-                    assertEquals("failures", parser.currentName());
-                    assertEquals(XContentParser.Token.START_ARRAY, parser.nextToken());
-
-                    for (int i = 0; i < shardInfo.getFailures().length; i++) {
-                        assertFailure(parser, shardInfo.getFailures()[i]);
-                    }
-                    assertEquals(XContentParser.Token.END_ARRAY, parser.nextToken());
-                }
+                assertEquals("failures", parser.currentName());
+                assertEquals(XContentParser.Token.START_ARRAY, parser.nextToken());
 
-                assertEquals(XContentParser.Token.END_OBJECT, parser.nextToken());
-                assertEquals(XContentParser.Token.END_OBJECT, parser.nextToken());
-                assertNull(parser.nextToken());
+                for (int i = 0; i < shardInfo.getFailures().length; i++) {
+                    assertFailure(parser, shardInfo.getFailures()[i]);
+                }
+                assertEquals(XContentParser.Token.END_ARRAY, parser.nextToken());
             }
+
+            assertEquals(XContentParser.Token.END_OBJECT, parser.nextToken());
+            assertEquals(XContentParser.Token.END_OBJECT, parser.nextToken());
+            assertNull(parser.nextToken());
+        }
+    }
+
+    public void testRandomShardInfoFromXContent() throws IOException {
```

Comment:
I changed the name to `testShardInfoWithFailureToXContent` and yes, in this case there are failures.